lib/Controller/PersonController.php

```diff
@@ -71,20 +72,30 @@ public function index() {
 		if ($userEnabled === 'true') {
 			$persons = $this->personMapper->findAll($this->userId);
 			foreach ($persons as $person) {
-				$cluster = [];
+				$personFaces = $this->faceMapper->findFacesFromPerson($this->userId, $person->getId(), $model);
+				if ($notGrouped === 'false' && count($personFaces) <= 1)
+					continue;
+
+				$limit = 14;
 				$faces = [];
-				$personFaces = $this->faceMapper->findFacesFromPerson($this->userId, $person->getId(), $model, 14);
 				foreach ($personFaces as $personFace) {
+					if ($limit-- === 0)
```

Comment:
This is quite horrible, but the query is practically just as fast, and by ignoring the single groups, the query gets smaller.